exec/vector/src/main/java/org/apache/drill/exec/vector/BitVector.java

```diff
@@ -328,7 +328,9 @@ public void splitAndTransferTo(int startIndex, int length, BitVector target) {
         lastButOneByte = byteIPlus1;
         bitsFromLastButOneByte = (byte)((lastButOneByte & 0xFF) >>> firstBitOffset);
 
-        final int lastByte = this.data.getByte(firstByteIndex + numBytesHoldingSourceBits);
+
+        final int lastByte = (8 - firstBitOffset >= length % 8) ?
+                0 : this.data.getByte(firstByteIndex + numBytesHoldingSourceBits);
```

Comment:
Check looks good. Can you please add the motivation of this check in the comment. Based on my understanding `length%8` gives the number of bits to be copied from last byte and `(8-firstBitOffset)` is the bits left to be copied from `lastButOneByte`. So if bits left to be copied is smaller than bits needed from last byte then we need to look at next byte in those cases.